assets/src/components/form/validations/index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * External imports
+ */
+import { __, _n, sprintf } from '@eventespresso/i18n';
+
+/**
+ * @function
+ * @param {mixed} value
+ * @return {string|null} error string on fail
+ */
+export const isFloat = value => {
+	// first convert to string
+	value = value + '';
+	// convert decimal to known format
+	value = value.replace( ',', '.' );
+	// strip all trailing zeros
+	value = value.replace( /0+$/, '' );
+	const lastChar = value.substr( value.length - 1 );
+	const float = (
+		// parsed value is same as original coerced to a number
+		parseFloat( value, 10 ) === +value &&
+		(
+			// no remainder but last character is a decimal
+			// ie: original value like 1.0
+			( value % 1 === 0 && lastChar === '.' ) ||
+			// has a remainder and last character is NOT a decimal
+			// ie: original value like 1.# where # is not 0
+			( value % 1 !== 0 && lastChar !== '.' )
+		)
+	);
```

Comment:
could much of this logic be replaced by something _based_ off of this?

```
function isFloat(n){
    return Number(n) === n && n % 1 !== 0;
}
```
The above function assumes the incoming value is a number literal though.  I'm not sure if `isFloat` is intended to handle "numeric" strings as well, IF so, you could likely also us [`isFinite`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isFinite) to help with determining whether a string is numeric or not.

Reply:
yes I too saw that version of this function on StackOverflow ;)

The issue with the above function is that `isFloat(1.0)` returns `false`
whereas many argue and expect that anything with a decimal value (even if it is zero) is a float.

`1.000000` is indeed an integer from a mathematical point of view because it is equivalent to `1` and this is how JavaScript itself approaches integers, but that conflicts with a more "everyday" approach to things like money values like `1.00` for dollars. That said, of course money could be validated using its own special set of rules.

So as convoluted as my approach may seem it allows both `1.0` and `1.2` to pass as floats, but not `1` or any other non-numerical value.

Reply:
> So as convoluted as my approach may seem it allows both 1.0 and 1.2 to pass as floats, but not 1 or any other non-numerical value.

Yup I pulled that from Stack Exchange :) More importantly, to the point of my argument for writing tests for these functions, that would have been immediately clear to me because your test could have demonstrated that for the purpose of this function `1.0` is considered a float.  As a reviewer, I'm not actually running the code (or testing it) I'm simply looking at the syntax and trying to spot any obvious logic flaws.